test/sharness/t0023-shutdown.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+#
+# Copyright (c) 2017 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test shutdown command"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+test_launch_ipfs_daemon
+
+test_expect_success "shutdown succeeds" '
+	ipfs shutdown
+'
+
+test_expect_success "daemon no longer running" '
+	test_expect_code  1 kill -0 $IPFS_PID
+'
+
```

Comment:
I would also test the daemon when started with the `--offline` flag.

When I added the test it failed.